src/main/java/seedu/storemando/model/item/ExpiryDate.java

```diff
@@ -3,31 +3,42 @@
 import static java.util.Objects.requireNonNull;
 import static seedu.storemando.commons.util.AppUtil.checkArgument;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
 /**
  * Represents an Item's expiry in the inventory manager.
  * Guarantees: immutable; is valid as declared in {@link #isValidExpiryDate(String)}
  */
 public class ExpiryDate {
 
-    private static final String SPECIAL_CHARACTERS = "!#$%&'*+/=?`{|}~^.-";
-    public static final String MESSAGE_CONSTRAINTS = "Expiry dates should be of the format local-part@domain "
+    public static final String NO_EXPIRY_DATE = "No Expiry Date";
+    private static final String SPECIAL_CHARACTERS = "-";
+    public static final String MESSAGE_CONSTRAINTS = "Expiry dates should be of the format YYYY-MM-DD "
         + "and adhere to the following constraints:\n"
-        + "1. The local-part should only contain alphanumeric characters and these special characters, excluding "
-        + "the parentheses, (" + SPECIAL_CHARACTERS + ") .\n"
-        + "2. This is followed by a '@' and then a domain name. "
-        + "The domain name must:\n"
-        + "    - be at least 2 characters long\n"
-        + "    - start and end with alphanumeric characters\n"
-        + "    - consist of alphanumeric characters, a period or a hyphen for the characters in between, if any.";
-    // alphanumeric and special characters
-    private static final String LOCAL_PART_REGEX = "^[\\w" + SPECIAL_CHARACTERS + "]+";
+        + "1. YYYY, MM and DD should only contain numeric characters and must be separated by a "
+        + SPECIAL_CHARACTERS
+        + " character.\n"
+        + "2. YYYY must:\n"
+        + "    - be exactly 4 digits long\n"
+        + "    - be a valid year\n"
+        + "3. MM must:\n"
+        + "    - be exactly 2 digits long\n"
+        + "    - be within the range [01, 12]\n"
+        + "4. DD must:\n"
+        + "    - be exactly 2 digits long\n"
+        + "    - be a valid date\n";
+    /*private static final String LOCAL_PART_REGEX = "^[\\w" + SPECIAL_CHARACTERS + "]+";
     private static final String DOMAIN_FIRST_CHARACTER_REGEX = "[^\\W_]"; // alphanumeric characters except underscore
     private static final String DOMAIN_MIDDLE_REGEX = "[a-zA-Z0-9.-]*"; // alphanumeric, period and hyphen
     private static final String DOMAIN_LAST_CHARACTER_REGEX = "[^\\W_]$";
     public static final String VALIDATION_REGEX = LOCAL_PART_REGEX + "@"
         + DOMAIN_FIRST_CHARACTER_REGEX + DOMAIN_MIDDLE_REGEX + DOMAIN_LAST_CHARACTER_REGEX;
+     */
 
     public final String value;
+    public final LocalDate date;
```

Comment:
Would it be better to name the field expiryDate to date? I think it would not be as ambiguous and would be easier to understand.

Reply:
alright!